examples/users_api_test.go

```diff
@@ -0,0 +1,56 @@
+package examples
+
+// TODO(dfurman): consider adding "realkentikapi" build tag so that examples are skipped by default
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/kentik/community_sdk_golang/kentikapi"
+)
+
+func TestUsersAPIExample(t *testing.T) {
```

Comment:
I see 3 ways to implement examples:
1. Testable examples (https://blog.golang.org/examples)
2. Tests, e.g. TestUsersAPIExample in examples/users_example_test.go
3. Multiple applications, e.g. `examples/users/users_example.go`

Details:

1. Testable examples do not fit our approach, because in order to be runnable , they need to contain `// Output` comment and we do not want to test outputs that come from external API.
2. IMO the best approach. They can be easily run all at once or one by one.
```
go test -tags examples ./examples/...  # run all
go test -tags examples ./examples/users_api_example.go   # run one file
```

Developers can also easily run all tests including examples (I added a build tag):
```
go test -tags examples ./...   # with examples
go test ./... # without examples
```

3. Package can contain only one main function to compile, so we would need to have following directory structure:

```
examples
│   ├── devices
│   │   └── devices_example.go
│   ├── users
│   │   └── users_example.go
```

Moreover there is no way to run all examples with one `go run ./examples/...` command. We would need to write a script for that.

I extracted `demonstrateUsersAPI` function, so that there is no `testing.T` function within the example code. It would look exactly the same in example as application. What do you think?

Reply:
This approach is using testing infrastructure for non-testing purpose, just to run things. This is misleading. But will work

Reply:
I added a note about that to readme:
> Note that they are placed in Go test files (e.g. _users_example_test.go_) to be easily runnable.